component-api/src/main/java/org/talend/sdk/component/api/service/discovery/DiscoverDatasetResult.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (C) 2006-2020 Talend Inc. - www.talend.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.talend.sdk.component.api.service.discovery;
+
+import java.util.List;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+@Data
+@AllArgsConstructor
+public class DiscoverDatasetResult {
+
+    private final Response response;
+
+    private final List<DatasetDescription> datasetDescriptionList;
+
+    @Data
+    @AllArgsConstructor
+    public final static class DatasetDescription {
+
+        private final String name;
+
+        private final String type;
+    }
```

Comment:
This seems to be typed `jdbc` connector focused...  We should have a more flex and extensible response content for future.

Reply:
I think that all datasets could have a type not only jdbc:
- in fileioS3 : type of file for instance (csv, json, parquet, ...)
- mscrm / salesforce / ... : entity
- ...
Could be something always displayed in list of dataset of the crawler.

For specific  informations I first thought to add a map but since it was not needed for jdbc I have preferred to not add it for the moment.

Reply:
Ok, for type but maybe you should add a kinda map for extra needed infos named metadata...
Should discuss this 1-to-1 tomorrow.

Reply:
That was my 1st thought but no need for jdbc so we'll se after.